internal/app/frontend/frontend_service.go

```diff
@@ -137,6 +137,106 @@ func doCreateBackfill(ctx context.Context, req *pb.CreateBackfillRequest, store
 	return backfill, nil
 }
 
+// UpdateBackfill updates a Backfill object, if present.
+func (s *frontendService) UpdateBackfill(ctx context.Context, req *pb.UpdateBackfillRequest) (*pb.Backfill, error) {
+	if req == nil {
+		return nil, status.Errorf(codes.InvalidArgument, "request is nil")
+	}
+	if req.Backfill == nil {
+		return nil, status.Errorf(codes.InvalidArgument, ".backfill is required")
+	}
+
+	backfill, ok := proto.Clone(req.Backfill).(*pb.Backfill)
```

Comment:
The update here should be done defensively, currently you perform the get, but use the backfill passed in as the source of truth.  Instead start with the backfill from the get, and then update each field which should be updated.  Eg, create time would be overwritten if the wrong value were sent here.

Reply:
Updated the code with:

```
	// Update generation here, because Frontend is used by GameServer only
	bfStored.Generation++
	bfStored.SearchFields = backfill.SearchFields
	bfStored.Extensions = backfill.Extensions
```